src/main/sensors/battery.c

```diff
@@ -232,16 +233,48 @@ static void batteryUpdateVoltageState(void)
         default:
             break;
     }
+
+}
+
+static void batteryUpdateLVC(timeUs_t currentTimeUs) 
+{   
+    if (batteryConfig()->lvcEnabled) {
+        if (voltageState == BATTERY_CRITICAL && !lowVoltageCutoff.enabled) {
+            lowVoltageCutoff.enabled = true;
+            lowVoltageCutoff.startTime = currentTimeUs;
+            lowVoltageCutoff.percentage = 100;
+        }
+        if (lowVoltageCutoff.enabled) {
+            if (currentTimeUs - lowVoltageCutoff.startTime < LVC_AFFECT_TIME) {
+                lowVoltageCutoff.percentage = 100 - ((currentTimeUs - lowVoltageCutoff.startTime) * (100 - batteryConfig()->lvcPercentage) / LVC_AFFECT_TIME); 
```

Comment:
You can convert the difference to millis first, then calculate percentage. But I'm not sure that it's the best way to do it ...
Using `millis()` is IMO bad idea - two time bases only make code harder to read ane 2^31us is enough for most calculations